StripeConnect/StripeConnect/Source/CustomFontSource.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  CustomFontSource.swift
+//  StripeConnect
+//
+//  Created by Chris Mays on 9/5/24.
+//
+
+import UIKit
+
+@_spi(PrivateBetaConnect)
+extension EmbeddedComponentManager {
+    
+    /// Use a `CustomFontSource` pass custom fonts embedded in your app's binary when initializing a
+    /// `EmbeddedComponentManager`.
+    /// - Seealso: https://docs.stripe.com/connect/get-started-connect-embedded-components#customize-the-look-of-connect-embedded-components
+    /// - Seealso: https://developer.apple.com/documentation/uikit/text_display_and_fonts/adding_a_custom_font_to_your_app
+    public struct CustomFontSource {
+        let family: String
+        let style: String?
+        let weight: String?
+        let src: FontSource?
+        
+        /**
+         Initializes a CustomFontSource from a base font and its original file URL
+         - Parameters:
+         - font: A custom font embedded into your app's binary
+         - fileUrl: The local file URL corresponding to the custom font
+         
+         - Note: The font's size does not impact the appearance of the component.
+         To adjust the font sizes used in components, use `EmbeddedComponentManager.Appearance`
+         
+         - Throws: Error if the font couldn't be loaded from the given URL
+         */
+        public init(font: UIFont, fileUrl: URL) throws {
+            guard fileUrl.isFileURL else {
+                throw FontLoadError.notFileURL
+            }
+            family = font.familyName
+            style = font.fontDescriptor.symbolicTraits.contains(.traitItalic) ? "italic" : nil
+            let traits = font.fontDescriptor.fontAttributes[.traits] as? [UIFontDescriptor.TraitKey: Any]
+            let weight = (traits?[.weight] as? CGFloat)
+                .map(UIFont.Weight.init(rawValue: ))
+            ?? .regular
+            guard let fontData = try? Data(contentsOf: fileUrl) else {
+                throw FontLoadError.noDataFoundAtURL
+            }
+            self.src = .init(fileType: fileUrl.pathExtension, encoding:fontData.base64EncodedString())
+            self.weight = weight.cssValue
+        }
+        
+        enum FontLoadError: Error {
+            case notFileURL
+            case noDataFoundAtURL
+        }
+    }
+    
+    struct FontSource: Encodable, Equatable {
+        let fileType: String
+        let encoding: String
```

Comment:
Encoding type or file type? If file type this feels like something the web view should validate and spit out an error if not correct since the file type support can be dynamic on web as well.